src/Syntax/Resolve.hs

```diff
@@ -198,13 +199,31 @@ lookupTy v = asks tyScope >>= lookupVar NotInScope v
 lookupTyvar :: MonadResolve m => Var Parsed -> m (Var Resolved)
 lookupTyvar v = asks tyvarScope >>= lookupVar NotInScope v
 
+resolveTele :: (MonadResolve m, Reasonable f p) => f p -> [TyConArg Parsed] -> m ([TyConArg Resolved], [(Var Parsed, Var Resolved)])
+resolveTele r (TyVarArg v:as) = do
+  v' <- tagVar v
+  (as, vs) <- resolveTele r as
+  pure (TyVarArg v':as, (v, v'):vs)
+resolveTele r (TyAnnArg v k:as) = do
+  v' <- tagVar v
+  ((as, vs), k) <- extendTyvar (v, v') $ 
+    (,) <$> resolveTele r as <*> reType r k
+  pure (TyAnnArg v' k:as, (v, v'):vs)
+resolveTele r (TyVisArg v k:as) = do
+  v' <- tagVar v
+  ((as, vs), k) <- extendTyvar (v, v') $ 
+    (,) <$> resolveTele r as <*> reType r k
+  pure (TyVisArg v' k:as, (v, v'):vs)
+resolveTele _ [] = pure ([], [])
+
 reExpr :: MonadResolve m => Expr Parsed -> m (Expr Resolved)
 reExpr r@(VarRef v a) = flip VarRef a <$> (lookupEx v `catchJunk` r)
 
 reExpr (Let vs c a) = do
-  let vars = map fst3 vs
+  let var (Binding v _ _ _) = v
```

Comment:
And here.

Reply:
Sorry, I don't know what I was thinking.